tools/installer/installer.py

```diff
@@ -0,0 +1,626 @@
+        self.log(f"Create group, {DEFAULT_CHEQD_USER} by default")
+        self.exec(f"addgroup {DEFAULT_CHEQD_USER} --quiet")
+
+        self.log(f"Create user, {DEFAULT_CHEQD_USER} by default")
+        self.exec(
+            f"adduser --system {DEFAULT_CHEQD_USER} --home {self.interviewer.home_dir} --shell /bin/bash --ingroup {DEFAULT_CHEQD_USER} --quiet")
+
+    def mkdir_p(self, dir_name):
+        try:
+            os.mkdir(dir_name)
+        except FileExistsError as err:
+            self.log(f"Directory {dir_name} already exists")
+
+    def setup_log_dir(self):
+        self.mkdir_p(self.cheqd_log_dir)
+        Path(os.path.join(self.cheqd_log_dir, "stdout.log")).touch()
+        self.exec(f"chown -R syslog:syslog {self.cheqd_log_dir}")
+
+    def setup_cosmovisor(self):
+        self.exec(f"wget -qO - {COSMOVISOR_BINARY}  | tar xz")
+        self.mkdir_p(self.cosmovisor_root_dir)
+        self.mkdir_p(os.path.join(self.cosmovisor_root_dir, "genesis"))
+        self.mkdir_p(os.path.join(self.cosmovisor_root_dir, "genesis/bin"))
+        self.mkdir_p(os.path.join(self.cosmovisor_root_dir, "upgrades"))
+        if not os.path.exists(os.path.join(DEFAULT_INSTALL_PATH, "cosmovisor")):
+            self.log(f"Moving cosmovisor binary to default installation directory")
+            shutil.move("./cosmovisor", DEFAULT_INSTALL_PATH)
+
+        self.log(f"Moving binary from {self.binary_path} to {self.cosmovisor_cheqd_bin_path}")
+        self.exec("sudo mv {} {}".format(self.binary_path, self.cosmovisor_cheqd_bin_path))
+
+        if not os.path.exists(os.path.join(self.cosmovisor_root_dir, "current")):
+            self.log(f"Making symlink current -> genesis")
+            os.symlink(os.path.join(self.cosmovisor_root_dir, "genesis"),
+                       os.path.join(self.cosmovisor_root_dir, "current"))
+
+        if not os.path.exists(os.path.join(DEFAULT_INSTALL_PATH, DEFAULT_BINARY_NAME)):
+            self.log(f"Making symlink to {self.cosmovisor_cheqd_bin_path}")
+            os.symlink(self.cosmovisor_cheqd_bin_path,
+                       os.path.join(DEFAULT_INSTALL_PATH, DEFAULT_BINARY_NAME))
+
+        self.log(f"Changing owner to {DEFAULT_CHEQD_USER} user")
+        self.exec(f"chown -R {DEFAULT_CHEQD_USER}:{DEFAULT_CHEQD_USER} {self.cosmovisor_root_dir}")
+
+    def untar_from_snapshot(self):
+        self.mkdir_p(self.cheqd_data_dir)
+        cmd = f"wget -c -O - {self.interviewer.snapshot_url}  | sudo -u {DEFAULT_CHEQD_USER} tar xzf - -C {os.path.join(self.cheqd_root_dir, 'data')}"
+        thread = threading.Thread(target=functools.partial(self.exec, cmd))
+        thread.start()
+        sec_counter = 0
+
+        # wait small period of time for waiting the command running
+        time.sleep(3)
+        while thread.is_alive():
+            time.sleep(60)
+            sec_counter += 60
+            self.log(f"Downloading is alive, it already took: {str(datetime.timedelta(seconds=sec_counter))}")
```

Comment:
Don't use this. Gives no idea whether it's still at download stage or untarring stage. Give progress bar for download and untar separately.